components/watcher/src/watcher_listener.rs

```diff
@@ -126,6 +125,23 @@ pub mod http_handlers {
             .body(resp))
     }
 
+    #[post("/discover")]
```

Comment:
This endpoint will be used for sending signed reply with end role obi inside. How about calling it `register`?